logstash-core/lib/logstash/config/lir_serializer.rb

```diff
@@ -64,7 +64,18 @@ def vertex_type(v)
       end
     end
     
+    def get_uuid(plugin_id)
+      if LogStash::PluginMetadata.exists?(plugin_id.to_s)
+        plugin_metadata = LogStash::PluginMetadata.for_plugin(plugin_id.to_s)
+        cluster_uuid = plugin_metadata&.get(:cluster_uuid)
+      end
+    end
+    
     def decorate_vertex(v, v_json)
+      uuid = get_uuid(v.id)
```

Comment:
this appears to be an entirely new change, so I feel like I'm reviewing a moving target here; what brought about the requirement to add this to the LIR serialisation? The LIR is an internal representation of the config's AST as compiled by the grammar, and injecting runtime metadata into it feels like a recipe for surprises later down the road.

Reply:
Hi @yaauie. Apologies for the confusion. This was a discussion [happening earlier in the PR discussion](https://github.com/elastic/logstash/pull/10576#issuecomment-498516575).

To avoid modifying the internal representation, do you think it would be better to try to do this lookup and insertion from [the API layer](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/api/commands/node.rb)?

Reply:
Yes, I believe that cluster-uuid is an API-level concern and should be injected at that layer if feasible (whereas the other things that are injected here in `decorate_vertex` are all guaranteed to be available as soon as the pipeline compiles, the `cluster_uuid` requires further initialisation and connection to external resources).

I've added a suggestion as to how this could be added in `PluginStats#report`.